src/model/chat.ts

```diff
@@ -147,15 +151,41 @@ export class ChatModel<
         }
 
         // Merge the delta into the chunk
-        const { content, function_call } = delta;
+        const { content, function_call, tool_calls } = delta;
         if (content) {
           chunk.choices[0].delta.content = `${chunk.choices[0].delta.content}${content}`;
         }
         if (function_call) {
-          chunk.choices[0].delta.function_call = deepMerge(
-            chunk.choices[0].delta.function_call,
-            function_call
-          );
+          const existingFunctionCall = chunk.choices[0].delta.function_call;
```

Comment:
Just so i understand, this is the code for streaming function calls right?

Reply:
this is for when `stream=true` on the request, and the assistant message is returns is a function or tool call, instead of just content. these are harder to do the merge+patch because the value that is being patched is more nested than just `delta.content` for the basic assistant string response.